### ISSUE TITLE
Gateway ServiceMonitor is invalid

### ISSUE THREAD

**Issue by nniehoff** [grafana/mimir#3688]
#### Describe the bug

When enabling the prometheus operator service monitors with `metaMonitoring.serviceMonitor.enabled: true` and the gateway with `gateway.enabledNonEnterprise: true` a ServiceMonitor is deployed to scrape the gateway for metrics at `/metrics`.  This endpoint is invalid

#### To Reproduce

0.  Setup the Prometheus Operator
1. Start Mimir Distributed Helm chart version 4.0.0-weekly.214
2. Set the values:

```yaml
gateway:
  enabledNonEnterprise: true

metaMonitoring:
  serviceMonitor:
    enabled: true
```

3.  Check the gateway logs and the Prometheus targets

#### Expected behavior

It is expected either metrics are collected from the gateway or no `ServiceMonitor` is created

#### Environment

- Infrastructure: K8s
- Deployment tool: helm

#### Additional Context

Checking the gateway logs, you can see the failed requests:

```
2022/12/10 23:24:47 [error] 9#9: *1 open() "/etc/nginx/html/metrics" failed (2: No such file or directory), client: 100.65.20.181, server: , request: "GET /metrics HTTP/1.1", host: "100.65.20.227:8080"
```

Check the prometheus targets `promethues/targets` and you'll notice the unhealthy target: `serviceMonitor/mimir/mimir-gateway/0 (0/3 up)`

**Comment by sheyabernstein** [comment]
Hi @dimitarvdimitrov ,

It looks like this issue has resurfaced. The bug was previously fixed in #3781, but recent changes seem to have reintroduced it.

**Comment by Rohlik** [comment]
Having the same error after migration to Mimir 3 (Helm chart 6.x).

**Comment by erikhandersson** [comment]
Having the same error with helm mimir-distributed-6.0.5 ( app 3.0.1 )

**Comment by Rohlik** [comment]
This bug should be reopened 🤔.

**Comment by NT-florianernst** [comment]
JFTR, no need to reopen this issue here as the service monitor creation for the nginx gateway with Helm chart v6.x was reported in #13674 and subsequently fixed by way of #13481.
